Update docs for newer devices and DeviceConfig (#614)
python-kasa/python-kasa: docs/source/discover.rst

.. py:module:: kasa.discover

Discovering devices
===================

.. contents:: Contents
   :local:

Discovery
*********

Discovery works by sending broadcast UDP packets to two known TP-link discovery ports, 9999 and 20002.

> 9999 is for legacy devices that didn't use strong encryption and 20002 is for newer devices that use different

Reviewer: ```suggestion
Port 9999 is used for legacy devices that do not use strong encryption and 20002 is for newer devices that use different
```

Author: Done

levels of encryption.

> If a device uses 20002 for discovery it will return some basic information with sensitive data hashed.

Reviewer: ```suggestion
If a device uses port 20002 for discovery, you will still obtain some basic information from the device, but you need to pass credentials for complete response.
```
Maybe something like this?

Author: Updated but with additional information.

To query or update the device requires authentication via :class:`Credentials <kasa.Credentials>` and if this is invalid or not provided it
will raise an :class:`AuthenticationException <kasa.AuthenticationException>`.

If discovery encounters an unsupported device when calling via :meth:`Discover.discover_single()  <kasa.Discover.discover_single>`
it will raise a :class:`UnsupportedDeviceException  <kasa.UnsupportedDeviceException>`.

> If discovery encounters a device when calling :meth:`Discover.discover() <kasa.Discover.discover>` you can provide a callback to the ``on_unsupported`` parameter

Reviewer: ```suggestion
If discovery encounters a device when calling :meth:`Discover.discover() <kasa.Discover.discover>`, you can provide a callback to the ``on_unsupported`` parameter
```

Author: Done

to handle these.

> Simple example script showing some functionality:

Reviewer: ```suggestion
Example:
```

Author: Done

.. code-block:: python

    import asyncio
    from kasa import Discover, Credentials

    async def main():
        device = await Discover.discover_single(
            "127.0.0.1",
            credentials=Credentials("myusername", "mypassword"),
            discovery_timeout=10
        )

        await device.update()  # Request the update
        print(device.alias)  # Print out the alias

        devices = await Discover.discover(
            credentials=Credentials("myusername", "mypassword"),
            discovery_timeout=10
        )
        for ip, device in devices.items():

>             await device.update()
> 

Reviewer: ```suggestion
            await device.update()
            print(device)

```
Let's print out some information so that this can be used directly.

Author: Done
